### PR TITLE
Arsenal Export

### PR DESCRIPTION
**Pull request by PabstMirror**
#131

### CODE REVIEW

**Comment by PabstMirror** [comment]
Use with https://github.com/BourbonWarfare/bwmf/blob/ec8ba282bbe9db7c958765d0ab642a14fbae4772/Loadouts/blankForArsenal.hpp
This will copy something like the following to clipboard, just paste it into the above loadout hpp and you are done.
```
// Camo set
#define CAMO_UNIFORM "CUP_I_TKG_Khet_Partug_08"
#define CAMO_VEST "CUP_V_OI_TKI_Jacket1_05"
#define CAMO_BACKPACK "CUP_B_TKG_Backpack_RPG"
#define CAMO_HEADGEAR "CUP_H_TKI_SkullCap_05"
// Rifle
#define RIFLE "rhs_weap_ak74m_gp25"
#define RIFLE_MAG "30Rnd_545x39_Mag_F:10"
#define RIFLE_ATTACHMENTS "rhs_acc_dtk"
#define AAR_ATTACHMENTS RIFLE_ATTACHMENTS
#define ALT_OPTICS "optic_Aco","rhsusf_acc_compm4","rhsusf_acc_eotech_xps3","CUP_optic_CompM2_Black","CUP_optic_TrijiconRx01_black","BWA3_optic_RSAS","RKSL_optic_RMR_MS19"
// GL Rifle
#define GLRIFLE "rhs_weap_ak74m_gp25"
#define GLRIFLE_MAG "30Rnd_545x39_Mag_F:10"
#define GLRIFLE_MAG_SMOKE "1Rnd_Smoke_Grenade_shell:2","1Rnd_SmokeRed_Grenade_shell:2"
#define GLRIFLE_MAG_HE "1Rnd_HE_Grenade_shell:5"
#define GLRIFLE_MAG_FLARE "UGL_FlareYellow_F:4"
// Carbine
#define CARBINE "rhs_weap_ak74m_gp25"
#define CARBINE_MAG "30Rnd_545x39_Mag_F:10"
// AR
#define AR "CUP_arifle_AKM"
#define AR_MAG "CUP_30Rnd_762x39_AK47_M:17"
// AT
#define AT "CUP_launch_NLAW"
#define AT_MAG "ACE_PreloadedMissileDummy_NLAW_CUP:1"
// MMG
#define MMG "CUP_arifle_AKM"
#define MMG_MAG "CUP_30Rnd_762x39_AK47_M:17"
// MAT
#define MAT "CUP_launch_NLAW"
#define MAT_MAG "ACE_PreloadedMissileDummy_NLAW_CUP:1"
#define MAT_MAG2 "ACE_PreloadedMissileDummy_NLAW_CUP:1"
#define MAT_OPTIC ""
```

File: addons/arsenalExport/XEH_postInit.sqf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#include "script_component.hpp"
+TRACE_1("",QUOTE(ADDON));
+
+[
+    missionnamespace,
+    "arsenalOpened",
+    {
+        INFO("arsenalOpened EH");
+        // call ACE_PREP_RECOMPILE;
```
Comment (AACO) [inline addons/arsenalExport/XEH_postInit.sqf]: Is the recompile still needed, or was it just for iterative testing?

File: addons/arsenalExport/functions/fnc_buttonClick.sqf [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                GVAR(loadout_atMags) pushBackUnique _x;
+            };
+        } forEach (magazines player);
+        systemChat format ["[Set %1]: %2 %3", _fncString, GVAR(loadout_at), GVAR(loadout_atMags)];
+    };
+case ("mat"): {
+        GVAR(loadout_mat) = secondaryWeapon player;
+        GVAR(loadout_matMags) = [];
+        {
+            if (_x in getArray (configFile >> "CfgWeapons" >> GVAR(loadout_mat) >> "magazines")) then {
```
Comment (AACO) [inline addons/arsenalExport/functions/fnc_buttonClick.sqf]: (goes for all mag checks) while we are pulling the values from the game, do we have to worry about casing issues?

File: addons/arsenalExport/functions/fnc_buttonClick.sqf [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        GVAR(loadout_rifleAttachments) = primaryWeaponItems player;
+        {
+            if (_x in getArray (configFile >> "CfgWeapons" >> GVAR(loadout_rifle) >> "magazines")) then {
+                GVAR(loadout_rifleMags) pushBackUnique _x;
+            };
+        } forEach (magazines player);
+        systemChat format ["[Set %1]: %2 %3 %4", _fncString, GVAR(loadout_rifle), GVAR(loadout_rifleMags), GVAR(loadout_rifleAttachments)];
+    };
+case ("glrifle"): {
+        GVAR(loadout_glrifle) = primaryWeapon player;
```
Comment (AACO) [inline addons/arsenalExport/functions/fnc_buttonClick.sqf]: We should pull GLRIFLE specific mags

File: addons/arsenalExport/functions/fnc_buttonClick.sqf [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        systemChat format ["[Set %1]: %2", _fncString, GVAR(loadout_glrifle)];
+    };
+case ("carbine"): {
+        GVAR(loadout_carbine) = primaryWeapon player;
```
Comment (AACO) [inline addons/arsenalExport/functions/fnc_buttonClick.sqf]: We should pull carbine mags, could be a completely different caliber

File: addons/arsenalExport/functions/fnc_buttonClick.sqf [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+};
+
+[77000, QGVAR(loadout_uniform), QGVAR(loadout_vest), QGVAR(loadout_backpack), QGVAR(loadout_headgear)] call _fnc_setDisplay;
+[77010, QGVAR(loadout_rifle), QGVAR(loadout_rifleMags), QGVAR(loadout_rifleAttachments)] call _fnc_setDisplay;
+[77020, QGVAR(loadout_glrifle)] call _fnc_setDisplay;
+[77030, QGVAR(loadout_carbine)] call _fnc_setDisplay;
+[77040, QGVAR(loadout_ar), QGVAR(loadout_arMags)] call _fnc_setDisplay;
+[77050, QGVAR(loadout_mmg), QGVAR(loadout_mmgMags)] call _fnc_setDisplay;
+[77060, QGVAR(loadout_at), QGVAR(loadout_atMags)] call _fnc_setDisplay;
+[77070, QGVAR(loadout_mat), QGVAR(loadout_matMags)] call _fnc_setDisplay;
```
Comment (AACO) [inline addons/arsenalExport/functions/fnc_buttonClick.sqf]: We should either globally hard code these, or store the idcs when they're generated, If someone changes the order/adds a button these values will be fucked

File: addons/arsenalExport/functions/fnc_export.sqf [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        _lines pushBack format ['#define GLRIFLE_MAG_HE "rhs_VOG25:5"'];
+        _lines pushBack format ['#define GLRIFLE_MAG_FLARE "rhs_VG40OP_red:4"'];
+    };
+    default {
+        _lines pushBack format ["// WARNING - Unknown GL Muzzle [%1->%2->%3]", GVAR(loadout_glrifle), _glMuzzle, _glMags];
+        _lines pushBack format ['#define GLRIFLE_MAG_SMOKE "TODO:2","TODO:2"'];
+        _lines pushBack format ['#define GLRIFLE_MAG_HE "TODO:5"'];
+        _lines pushBack format ['#define GLRIFLE_MAG_FLARE "TODO:4"'];
+    };
+};
```
Comment (AACO) [inline addons/arsenalExport/functions/fnc_export.sqf]: Pretty sound logic, with the massCompat mod all RU should have access to GRD40s, all NATO should have the vanilla smokes

File: addons/arsenalExport/$PBOPREFIX$ [hunk 7]
```diff
@@ -0,0 +1 @@
+z\potato\addons\arsenalExport
```
Comment (AACO) [inline addons/arsenalExport/$PBOPREFIX$]: Should this be independent of the assignGear framework? They're pretty hand in hand. With that said making assign gear more complicated may not be the best strategy.....

Comment (PabstMirror) [inline addons/arsenalExport/$PBOPREFIX$]: Could go either place.
There are no code dependencies so I like having it separate which leaves each module cleaner.

File: addons/arsenalExport/XEH_postInit.sqf (hunk 1 above)
Comment (PabstMirror) [inline addons/arsenalExport/XEH_postInit.sqf]: Just for testing

**Pull request merged**